core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -651,6 +727,36 @@ class GroupMetadataManager(brokerId: Int,
     info(s"Removed $offsetsRemoved expired offsets in ${time.milliseconds() - startMs} milliseconds.")
   }
 
+  def handleTxnCompletion(producerId: Long, completedPartitions: Set[Int], isCommit: Boolean) {
+    val pendingGroups = removeProducerGroupsBelongingToPartitions(producerId, completedPartitions)
+    pendingGroups.foreach { case (groupId) =>
+      getGroup(groupId) match {
+        case Some(group) => group synchronized {
+          group.completePendingTxnOffsetCommit(producerId, isCommit)
+        }
+        case _ =>
+          info(s"Group $groupId has moved away from $brokerId after transaction marker was written but before the " +
+            s"cache was updated. The cache on the new group owner will be updated instead.")
+      }
+    }
+  }
+
+  private def addProducerGroup(producerId: Long, groupId: String) = openGroupsForProducer synchronized {
+    openGroupsForProducer.getOrElseUpdate(producerId, mutable.Set.empty[String]).add(groupId)
+  }
+
+  private def removeProducerGroup(producerId: Long, groupId: String) = openGroupsForProducer synchronized {
+    openGroupsForProducer.getOrElseUpdate(producerId, mutable.Set.empty[String]).remove(groupId)
```

Comment:
I think I have misunderstood what you are asking. There is a check right below which removes the map entry for the producerId if the set of open groups is empty.